pkg/backend/httpstate/client/client.go

```diff
@@ -1138,6 +1139,43 @@ func (pc *Client) GetCapabilities(ctx context.Context) (*apitype.CapabilitiesRes
 	return &resp, nil
 }
 
+func getSearchPath(orgName string) string {
+	return fmt.Sprintf("/api/orgs/%s/search/resources", orgName)
+}
+
+func getNaturalLanguageSearchPath(orgName string) string {
+	return fmt.Sprintf("/api/orgs/%s/search/resources/parse", orgName)
+}
+
+// Pulumi Cloud Search Functions
+func (pc *Client) GetSearchQueryResults(
+	ctx context.Context, orgName string, queryParams *apitype.PulumiQueryRequest,
+) (*apitype.ResourceSearchResponse, error) {
+	var resp apitype.ResourceSearchResponse
+	err := pc.restCall(ctx, http.MethodGet, getSearchPath(orgName), *queryParams, nil, &resp)
+	if err != nil {
+		return nil, fmt.Errorf("querying search failed: %w", err)
+	}
+	return &resp, nil
+}
+
+func (pc *Client) GetNaturalLanguageQueryResults(
+	ctx context.Context, orgName string, queryString string,
+) (*apitype.PulumiQueryResponse, error) {
+	var resp apitype.PulumiQueryResponse
+	var cleanedResponse apitype.PulumiQueryResponse
```

Comment:
style nit: these can be declared together.
```suggestion
	var resp, cleanedResponse apitype.PulumiQueryResponse
```